### PR TITLE
Fs2 0.10.0 m3

### PR DESCRIPTION
**Pull request by antikantian**

### CODE REVIEW

**Comment by guillaumebort** [comment]
I saw the tests failing once. After a relaunch it succeeded. So there is probably a race condition somewhere. I don't see where because it looks like the changes are trivial, but perhaps there is some semantic changes between Task and IO... or there is a bug in the impl. I will test it more to see if I can find a way to reproduce it.

**Comment by guillaumebort** [comment]
I can't make the test fail in my own computer. In your pull request can you change the default timeout from 5 seconds to 30 seconds in `Tests.scala` (changing all `atMost: Duration = 5 seconds`). I just want to be sure that the call are really locked, not just slower for any reason.

**Comment by antikantian** [comment]
Hmm, yeah, I just noticed that the tests sporadically fail w/travis, which I couldn't reproduce on my own computer either.

**Comment by guillaumebort** [comment]
Ok actually I managed to make the test fail the same way on master as well. So it is probably not related to this pull request. I will do the complete review then.

File: core/src/main/scala/Server.scala [hunk 1]
```diff
@@ -13,7 +13,9 @@ import scala.util.{ Try }
 import scala.concurrent.{ExecutionContext, Promise, Future }
 import scala.concurrent.duration._
 
-import fs2.{ Task, Strategy }
+import cats.Eval
+import cats.effect.IO
+import cats.effect.implicits._
```
Comment (guillaumebort) [inline core/src/main/scala/Server.scala]: This is unused.

File: core/src/main/scala/internal/NettySupport.scala [hunk 2]
```diff
@@ -1,41 +1,20 @@
...
-  HttpVersion => NettyHttpVersion,
-  HttpResponseStatus,
-  HttpRequest,
-  HttpUtil }
-
-import scala.concurrent.{ Future, Promise }
-import scala.collection.mutable.{ ListBuffer }
+import scala.concurrent.ExecutionContext
+
+import cats._
```
Comment (guillaumebort) [inline core/src/main/scala/internal/NettySupport.scala]: `cats._` is unused.

File: core/src/main/scala/internal/NettySupport.scala [hunk 3]
```diff
@@ -1,41 +1,20 @@
...
+import cats.effect.IO
+import fs2.{Catenable, Chunk, Pull, Segment, Sink, Stream, async}
+import io.netty.channel.{Channel, ChannelFuture, ChannelHandlerContext, SimpleChannelInboundHandler}
+import io.netty.handler.logging.{LogLevel, LoggingHandler}
+import io.netty.util.concurrent.GenericFutureListener
+import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.handler.codec.http.{DefaultHttpContent, DefaultHttpRequest, DefaultHttpResponse, DefaultLastHttpContent, HttpClientCodec, HttpContent, HttpContentDecompressor, HttpMessage, HttpObject, HttpRequest, HttpRequestDecoder, HttpResponse, HttpResponseEncoder, HttpResponseStatus, HttpUtil, LastHttpContent, HttpMethod => NettyHttpMethod, HttpVersion => NettyHttpVersion}
```
Comment (guillaumebort) [inline core/src/main/scala/internal/NettySupport.scala]: Can you please keep the existing formatting? This line is very long.

File: core/src/main/scala/internal/NettySupport.scala [hunk 4]
```diff
@@ -1,41 +1,20 @@
...
+import scala.concurrent.{Future, Promise}
+import scala.collection.mutable.ListBuffer
 import collection.JavaConverters._
 
-import fs2.{ Task }
-
+import io.netty.util
```
Comment (guillaumebort) [inline core/src/main/scala/internal/NettySupport.scala]: This is useless.

File: core/src/main/scala/internal/NettySupport.scala [hunk 5]
```diff
@@ -444,72 +436,65 @@ private[http] object NettySupport {
...
-                      // When user code finished to consume this stream, we need
-                      // to drain the remaining content if the eos has not been
-                      // reached yet.
-                      onFinalize(for {
-                        fullyRead <- eosReached.get
-                        _ <- if(fullyRead) Task.now(()) else contentStream.takeWhile(_.isDefined).drain.run
-                        _ <- if(client) permits.increment else permits.decrement
-                      } yield ())
-                }
+              Stream.eval(readers.tryDecrement).flatMap {
```
Comment (guillaumebort) [inline core/src/main/scala/internal/NettySupport.scala]: Why did you remove the existing comments?

File: core/src/main/scala/internal/NettySupport.scala (hunk 3 above)
Comment (antikantian) [inline core/src/main/scala/internal/NettySupport.scala]: Ugh, sorry, that line is embarrassingly long.  Must have been an IntelliJ auto-format that I didn't catch, since IntelliJ collapses all my imports.

File: core/src/main/scala/internal/NettySupport.scala (hunk 5 above)
Comment (antikantian) [inline core/src/main/scala/internal/NettySupport.scala]: Sorry again, when I refactored that, I made a mental note to put those comments back, but it slipped my mind.

**Comment by antikantian** [comment]
I'll get those changes done in an hour or two.  Thanks!

**Comment by guillaumebort** [comment]
Thanks!
